detectable/src/main/java/com/synopsys/integration/detectable/detectables/yarn/YarnTransformer.java

```diff
@@ -39,74 +41,52 @@
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
     public static final String STRING_ID_NAME_VERSION_SEPARATOR = "@";
     private final ExternalIdFactory externalIdFactory;
+    private final Set<StringDependencyId> unMatchedDependencies = new HashSet<>();
```

Comment:
I don't love having this here. I know you are only doing it to prevent duplicate logs but do we have that precedent anywhere else? Should other package managers try to avoid logging duplicate missing error messages?

Reply:
It's worse for yarn because of the way it's processing the lock file (once per workspace). I've created IDETECT-2624 for this... hopefully we can address that soon and then remove that.